index.bs

```diff
@@ -643,12 +656,34 @@ The {{VRDisplay}} associated with this event.
 {{VRDisplayEventReason}} describing why this event has has been fired.
 
 
-## Window Interface extension ## {#interface-window}
+## VREventHandlers Interface ## {#interface-vreventhandlers}
 
 <pre class="idl">
-partial interface Window {
+[NoInterfaceObject]
+interface VREventHandlers {
   attribute EventHandler onvrdisplayconnect;
   attribute EventHandler onvrdisplaydisconnect;
+  attribute EventHandler onvrdisplaynavigate;
+};
+</pre>
+
+User agents implementing this specification MUST provide the following new DOM events. The corresponding events must be of type {{VRDisplayEvent}} and must fire on the {{vr}} object. Registration for and firing of the events must follow the usual behavior of DOM4 Events.
+
+<dfn event for="VREventHandlers" id="vreventhandlers-vrdisplayconnect-event">onvrdisplayconnect</dfn>
+A user agent MAY dispatch this event type to indicate that a {{VRDisplay}} has been connected.
+
+<dfn event for="VREventHandlers" id="vreventhandlers-vrdisplaydisconnect-event">onvrdisplaydisconnect</dfn>
+A user agent MAY dispatch this event type to indicate that a {{VRDisplay}} has been disconnected.
+
+<dfn event for="VREventHandlers" id="vreventhandlers-vrdisplaynavigate-event">onvrdisplaynavigate</dfn>
+A user agent MAY dispatch this event type to indicate that the current page has been navigated to from a page that was actively presenting VR content to the {{VRDisplay}}. The current page can call {{requestPresent()}} in response to this event in order to stay in VR presentation mode.
```

Comment:
We need to know when to fire this event since this will be one of the indications that we should stay in VR presentation mode.